### ISSUE TITLE
ltss won't load after update to ha-core 2023.3.2

### ISSUE THREAD

**Issue by agittins** [freol35241/ltss#56]
I'm running supervised on KVM, and timeseriesdb in a container outside of that VM, all of which has been running well for months.

I just updated from ha core 2023.2.something to 2023.3.2, using the web-ui.

I got errors at that point about not being able to satisfy the requirement for sqlalchemy >=1 and <2, so I updated ltss then from v1.x.x to the current release 2.0.1.

After rebooting the full VM, I get a notification that the ltss config is invalid:
![image](https://user-images.githubusercontent.com/752405/224023855-8b008998-aca0-42fb-8512-128e1fbf2e0f.png)

And the logs show this error:
```
Logger: homeassistant.setup
Source: setup.py:213
First occurred: 23:19:59 (1 occurrences)
Last logged: 23:19:59

Setup failed for custom integration ltss: Unable to import component: cannot import name 'with_metaclass' from 'sqlalchemy.util' (/usr/local/lib/python3.10/site-packages/sqlalchemy/util/__init__.py)
```

I tried (through the hacs gui) deleting and then downloading the ltss package again, but with the same result.

The docker container that supervisor is running for homeassistant is `74baa42900dd   ghcr.io/home-assistant/qemux86-64-homeassistant:2023.3.2` and running `python --version` in it reports `Python 3.10.10`. 

If I also run `pip show sqlalchemy` in that same container, I get:
```
# pip show sqlalchemy
Name: SQLAlchemy
Version: 2.0.5.post1
Summary: Database Abstraction Library
Home-page: https://www.sqlalchemy.org
Author: Mike Bayer
Author-email: mike_mp@zzzcomputing.com
License: MIT
Location: /usr/local/lib/python3.10/site-packages
Requires: greenlet, typing-extensions
Required-by: GeoAlchemy2, pygtfs
```

My `configuration.yaml` contains:
```
# AJG 2022-07-04 Now using PostgreSQL and TimescaleDB for history etc
recorder:                                                            
  db_url: !secret postgres_url                                       
ltss:                                                                
  db_url: !secret postgres_url                                       
  chunk_time_interval: 2592000000000                                 
```
(the `chunk_time_interval` was added today after the update to 2.0.1 and reading the new readme).

I couldn't see any issues searching the sqlalchemy tracker for with_metaclass, so I'm not really sure where to go from here. Any pointers to help work out what's happening?

**Comment by agittins** [comment]
Gah, sorry - somehow I missed #50 I'll try the alpha.

**Comment by freol35241** [comment]
Hi, this is a problem with HA that will not update to the required versions of python dependencies it seems. In this case it is GeoAlchemy2 that is the offender. Check you version of GeoAlchemy2 and compare with the requirements in LTSS v2.0.1.

Also, see this comment: https://github.com/freol35241/ltss/issues/50#issuecomment-1456528287

**Comment by agittins** [comment]
Thanks for the pointer! From inside the container I get:
```
# pip show geoalchemy2
Name: GeoAlchemy2
Version: 0.13.1
Summary: Using SQLAlchemy with Spatial Databases
Home-page: https://geoalchemy-2.readthedocs.io/en/latest/
Author: Eric Lemoine
Author-email: eric.lemoine@gmail.com
License: MIT
Location: /usr/local/lib/python3.10/site-packages
Requires: packaging, SQLAlchemy
Required-by:
```

Which looks like it should satisfy the `>0.13,<1.0` in the manifest requirements. However as that comment pointed out, I can see in `deps/` that it still has a copy of 0.8.5.

I renamed the deps folder and restarted, then updated back to 2.0.1 from the alpha (having now clicked what the timeline was there! 😅 ).

Now it all looks good. So I'm wondering if the deps were downloaded but maybe the deps folder in HA was masking the newly available package? I notice that after restarting the deps folder was re-created, but there is nothing inside it. I wonder if the deps folder was deprecated at some point but missed getting cleaned up?

I'm sure I saw another issue recently where HA failed to honour the dependencies of a hacs unit during upgrade, I wonder if there's a bug further up in HA?

**Comment by freol35241** [comment]
Good that you got this working 👍 Yep, definitely seems like an upstream bug somewhere. Do you by any chance have a link to the issue you mention?

**Comment by agittins** [comment]
Unfortunately, no - I can't even remember what integration is was. Actually come to think of it the issue might have been with the packager using "dependencies" instead of "requirements", which IIRC had to do with load order rather than installation dependencies. Not sure. Anyway, thanks for the help and for the whole timescaledb integration!
